pkg/segment/query/iqr/intermediateQueryResult_test.go

```diff
@@ -406,6 +406,46 @@ func Test_MergeIQRs(t *testing.T) {
 	assert.Equal(t, 0, iqr3.NumberOfRecords())
 }
 
+func Test_MergeIQR_withNotSetIQRs(t *testing.T) {
+	less := func(a, b *Record) bool {
+		aVal, err := a.ReadColumn("col1")
+		assert.NoError(t, err)
+
+		bVal, err := b.ReadColumn("col1")
+		assert.NoError(t, err)
+
+		return aVal.CVal.(string) < bVal.CVal.(string)
+	}
+
+	_, firstExhaustedIndex, err := MergeIQRs([]*IQR{NewIQR(0), NewIQR(0), NewIQR(0)}, less)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, firstExhaustedIndex)
+
+	iqr1 := NewIQR(0)
+	err = iqr1.AppendKnownValues(map[string][]utils.CValueEnclosure{
+		"col1": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "a"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "e"},
+		},
+	})
+
+	assert.NoError(t, err)
+	_, firstExhausted, err := MergeIQRs([]*IQR{NewIQR(0), iqr1}, less)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, firstExhausted)
+}
+
+func Test_Mode_AfterAppendRRC(t *testing.T) {
+	iqr := NewIQR(0)
+	assert.Equal(t, notSet, iqr.mode)
+
+	encodingToSegKey := map[uint16]string{1: "segKey1"}
+
+	err := iqr.AppendRRCs([]*utils.RecordResultContainer{}, encodingToSegKey)
+	assert.NoError(t, err)
+	assert.Equal(t, withRRCs, iqr.mode)
+}
```

Comment:
fyi: I think this is already tested in `Test_AppendRRCs`, unless this test is specifically about about appending an empty list of RRCs

Reply:
Yeah, this was added just to ensure that if appendRRC is called on a notSet IQR, it should be set to withRRC mode.

Reply:
Ok, then something like `Test_Mode_AfterEmptyAppendRRC` would be a little better, but that's a small nit